core/model/src/geometry_sampled_field.cpp

```diff
@@ -100,10 +100,16 @@ static void setPixelsToValues(QImage &img, const std::vector<QRgb> &values) {
 
 static std::vector<QRgb> setImagePixelsNative(
     QImage &img, const libsbml::SampledField *sampledField,
-    const std::vector<const libsbml::SampledVolume *> &sampledVolumes) {
+    const std::vector<const libsbml::SampledVolume *> &sampledVolumes,
+    std::vector<QRgb> importedcompartmentColours = {}) {
   std::vector<QRgb> colours;
   colours.reserve(sampledVolumes.size());
   auto values = common::stringToVector<QRgb>(sampledField->getSamples());
+  if (importedcompartmentColours.size()) {
```

Comment:
`setImagePixelsNative` is used for importing sampled fields as we (currently) export them, where each value is a QRgb colour. Here you're replacing each of these values with the imported colour assuming the values are `0, 1, 2`, but typically the values are something like `0xffff0000`, i.e. a large unsigned int, so `importedcompartmentColours[values[i]]` will probably segfault.

It would be easier to leave this function as it is, and instead modify `template <typename T>
static std::vector<QRgb> setImagePixels` to take this `importedcompartmentColours` as an argument and use it instead of the default indexed colours if valid (see the Import section in https://github.com/spatial-model-editor/spatial-model-editor/issues/710#issuecomment-1275768189)